core/trino-main/src/main/java/io/trino/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -1847,7 +1847,8 @@ public static FaultTolerantDistributedStagesScheduler create(
                             inputBucketToPartition.getBucketNodeMap(),
                             remainingTaskRetryAttemptsOverall,
                             taskRetryAttemptsPerTask,
-                            maxTasksWaitingForNodePerStage);
+                            maxTasksWaitingForNodePerStage,
+                            scheduledExecutorService);
```

Comment:
I don't recommend using the `scheduledExecutorService` as it is single threaded and intended to be used only for lightweight task scheduling. I would probably recommend going with the `@ForQueryExecution` executor